test/xbgpu/test_engine.py

```diff
@@ -1086,7 +1157,7 @@ def heap_factory(batch_index: int, present: np.ndarray) -> list[spead2.send.Heap
             prom_diff=prom_diff,
             actual_sensor_updates=actual_sensor_updates,
             first_timestamp=first_timestamp,
-            last_timestamp=last_timestamp,
+            last_timestamp=steady_state_timestamps[-1],
```

Comment:
This no longer matches the documentation for this parameter of `verify_beam_sensors`. It looks like you've hijacked this parameter and used it for a different purpose. You should probably change the parameter names and the documentation of them.